app/Http/Controllers/PostController.php

```diff
@@ -87,4 +87,34 @@ public function destroy($id)
 
         return back()->with('success', __('post.error'));
     }
+
+    /**
+     * Share a post.
+     *
+     * @param  \Illuminate\Http\PostRequest  $request
+     * @param Int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function share(PostRequest $request, $id)
+    {
+        $post = Post::findOrFail($id);
+        $sharePostId = $post->share_from_post_id;
+
+        $data = $this->postService->getPostData($request);
+        $data['user_id'] = auth()->id();
+
+        if (is_null($sharePostId)) {
+            $data['share_from_post_id'] = $id;
+        } else {
+            $data['share_from_post_id'] = $sharePostId;
+        }
```

Comment:
```suggestion
            $data['share_from_post_id'] = is_null($sharePostId) ? $id : $sharePostId;
```
mà đưa mấy cái xử lý logic này vào service cũng đc nhé